internal/edgedbtypes/uuid.go

```diff
@@ -42,16 +43,25 @@ func (id UUID) MarshalText() ([]byte, error) {
 	return []byte(id.String()), nil
 }
 
+var errMalformedUUID = errors.New("malformed edgedb.UUID")
+
 // UnmarshalText unmarshals the id from a string.
 func (id *UUID) UnmarshalText(b []byte) error {
 	s := string(b)
+	if len(s) != 36 {
```

Comment:
I'd suggest different logic here:

`if len(s) < 32 || len(s) > 36 { return errMalformedUUID }`

This is what we have in EdgeDB: https://github.com/MagicStack/py-pgproto/blob/90ad4154a0aa6923e2f97d3d3d779c4b830ff415/uuid.pyx#L62-L75

The idea is that uuid string serilization is kinda arbitrary with respect to where the dashes are. Ultimately uuids are 128bit integers, and dashes are there to make them more readable to humans. E.g. `B7B69960-26D1-4BD4-8182876A1CC38BEE` can be just as valid as `B7B69960-26D1-4BD4-8182-876A1CC38BEE`.

Python also uses the same approach in its stdlib: https://github.com/python/cpython/blob/04f6fbb6969e9860783b9ab4dc24b6fe3c6dab8d/Lib/uuid.py#L173-L178

Reply:
That said I don't think we need to handle `urn:` and `uuid:` prefixes just yet.

Reply:
changed to remove all `-`s and then assert `len(32)`